src/services/rest/RestService.ts

```diff
@@ -0,0 +1,148 @@
+              private $q: ng.IQService,
+              private $location: ng.ILocationService,
+              private socketBindingService: SocketBindingService) {
+
+    // TODO somehow discover installed rest binding better than this?
+    socketBindingService.getDefaultSocketBindingGroup().then((result: ISocketBinding []) => {
+      angular.forEach(result, (value) => {
+        if (value.name === "rest") {
+          this.restBinding = value;
+        }
+      });
+    });
+  }
+
+  executeCacheGet(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("get", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDelete(cacheTarget: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: undefined
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheDeleteKey(cacheTarget: string, key: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: undefined,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("delete", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCachePut(cacheTarget: string, key: string, data: any): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: data,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("put", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCachePost(cacheTarget: string, key: string, data: any): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: data,
+      cache: cacheTarget,
+      urlSuffix: "/".concat(key)
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("post", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  executeCacheQuery(cacheTarget: string, query: string): ng.IPromise<any> {
+    let request: IRestRequest = {
+      jsonPayload: {query: query},
+      cache: cacheTarget,
+      urlSuffix: "?action=search"
+    };
+    let url: string = this.generateUrl(request);
+    return this.executeHelper("post", url, request.jsonPayload, this.getDefaultHeaders());
+  }
+
+  clearGetCache(): void {
+    this.$cacheFactory.get("$http").removeAll();
+  }
+
+  private executeHelper(type: string, url: string, data: any, headers?: ng.IRequestShortcutConfig): ng.IPromise<any> {
+    let deferred: ng.IDeferred<any> = this.$q.defer<any>();
+    let response: IHttpPromise;
+    switch (type) {
+      case "post":
```

Comment:
This would be much nicer if we could use enums introduced in Typescript 2.4. I know this PR isn't the place for such an upgrade, but we really should try to keep on top of updating our dependency versions as the longer we postpone such tasks, the harder these tasks become.

Reply:
+1